syntax = "proto3";

message CreateLocationRequest {
    int32 person_id = 1;
    string longitude = 2;
    string latitude = 3;
    string creation_time = 4;
}

message GetLocationRequest {
    int32 location_id = 1;
}

message ListLocationsRequest {
    int32 person_id = 1;
    string start_date = 2;
    string end_date = 3;
}

message Location {
    int32 location_id = 1;
    int32 person_id = 2;
    string longitude = 3;
    string latitude = 4;
    string creation_time = 5;
}

message ListLocationsResponse {
    repeated Location locations = 1;
}

service LocationService {
    rpc CreateLocation(CreateLocationRequest) returns (Location);
    rpc GetLocation(GetLocationRequest) returns (Location);
    rpc ListLocations(ListLocationsRequest) returns (ListLocationsResponse);
}
